### PR TITLE
Add wallet set up to Onboarding Wizard

### PR DESCRIPTION
**Pull request by pferreira101**
This PR adds 2 new steps to the Onboarding Wizard: one to ask if the user wants to add a test wallet and another with the HD Wallet creation form. One of the wallets must be added.

Why:
- This is part of this issue https://github.com/iron-wallet/iron/issues/311.

How:
- Add 2 new components/steps (`CreateTestWalletStep` and `AddHDWalletStep`);
- Stop creating the default wallet on init;
- Only create test wallet on onboarding close, if the user opted-in or didn't add an HD wallet;

Preview:
- `CreateTestWalletStep` where the user can choose to opt-out or input another valid mnemonic:
first

- `AddHDWalletStep` - HD Wallet creation form. The next step button (arrow at the right) is disabled if the user opted-out at the previous step:
blockedNext

- After saving the HD wallet, a success message is displayed below and the next step button becomes enabled:
last

Note: if the onboarding is skipped, a test wallet is created nonetheless (using the last valid mnemonic inputed at `CreateTestWalletStep`).

### CODE REVIEW

File: crates/rpc/src/lib.rs [hunk 1]
```diff
@@ -129,7 +129,11 @@ impl Handler {
 
     async fn accounts(_: Params, _: Ctx) -> jsonrpc_core::Result<serde_json::Value> {
         let wallets = Wallets::read().await;
-        let address = wallets.get_current_wallet().get_current_address().await;
+        let address = wallets
+            .get_current_wallet()
+            .ok_or(jsonrpc_core::Error::internal_error())?
```
Comment (naps62) [inline crates/rpc/src/lib.rs]: the error coming from here should include more info. not just an "internal error"

File: crates/wallets/src/error.rs [hunk 2]
```diff
@@ -41,6 +41,9 @@ pub enum Error {
 
     #[error(transparent)]
     ParseInto(#[from] std::num::ParseIntError),
+
+    #[error("No current wallet available")]
+    NoCurrentWallet,
```
Comment (naps62) [inline crates/wallets/src/error.rs]: would be more clear to say "No wallets" rather than "no current wallet" right?

File: gui/src/components/Onboarding/Steps.tsx [hunk 3]
```diff
@@ -1,5 +1,20 @@
...
+  FormControlLabel,
+  Grid,
+  Link,
+  Stack,
+  TextField,
+  Typography,
+} from "@mui/material";
+import { invoke } from "@tauri-apps/api/tauri";
+import { ChangeEvent, useState } from "react";
+import { type FieldValues } from "react-hook-form"; // TODO: check this import
```
Comment (naps62) [inline gui/src/components/Onboarding/Steps.tsx]: what's the problem with this import?

File: gui/src/components/Onboarding/Carousel.tsx [hunk 4]
```diff
@@ -33,6 +33,9 @@ export function OnboardingCarousel({
   const maxSteps = steps.length;
   const step = steps[activeStep];
 
+  const disableNext =
+    activeStep == 3 && !(formData.createTestWallet || formData.addedHDWallet);
```
Comment (naps62) [inline gui/src/components/Onboarding/Carousel.tsx]: it's not at all clear to me what this variable is suposed to represent

File: gui/src/components/Onboarding/Steps.tsx [hunk 5]
```diff
@@ -101,6 +124,118 @@ function LiveBlockchainsStep({
   );
 }
 
+function CreateTestWalletStep({
+  formData,
+  setFormData,
+}: {
+  formData: WizardFormData;
+  setFormData: React.Dispatch<React.SetStateAction<WizardFormData>>;
+}) {
```
Comment (naps62) [inline gui/src/components/Onboarding/Steps.tsx]: please move these types to a `interface CreateTestWalletStepProps` above

It's personal preference, I know, but I find it really hard to parse function headers like this

**Review by naps62 — COMMENTED** [review]
nice progress.  can you add some screenshots to the PR description?

File: crates/rpc/src/lib.rs (hunk 1 above)
Comment (pferreira101) [reply crates/rpc/src/lib.rs]: Agree, but given the return type this was the only error that I found suitable

File: gui/src/components/Onboarding/Steps.tsx (hunk 3 above)
Comment (pferreira101) [reply gui/src/components/Onboarding/Steps.tsx]: Forgot to delete this comment 😅 
But this was due to the signature on the `HDWalletForm` `onSubmit` prop being `Props.onSubmit: (data: FormType) => void` and I was passing it this `onSubmit: (params: FieldValues) => void`

File: gui/src/components/Onboarding/Carousel.tsx (hunk 4 above)
Comment (pferreira101) [reply gui/src/components/Onboarding/Carousel.tsx]: Renamed it to `disableNextStepButton`. It is used to determine if the carousel control button (on the right side) should be disabled. I'm not very happy with this implementation, especially the `activeStep == 3` which means that we are on the HD Wallet creation step, but I don't see a more idiomatic way to do it 😕

File: gui/src/components/Onboarding/Steps.tsx (hunk 5 above)
Comment (pferreira101) [reply gui/src/components/Onboarding/Steps.tsx]: Addressed this [here](https://github.com/Subvisual-Academy/iron/pull/1/commits/f2e5a3e7a8e365a91c04c4d9fc020991cc1ec179).
This was happening at two more components which reenforces your point.
Looks much clearer and easier to read now, thanks for the suggestion 🙌

File: crates/rpc/src/lib.rs (hunk 1 above)
Comment (pferreira101) [reply crates/rpc/src/lib.rs]: What do you think of [this](https://github.com/Subvisual-Academy/iron/pull/1/commits/55751c283da37ba4b663682a66e75d092291c3b4)
Wasn't aware of `impl From<Error> for jsonrpc_core::Error`.

File: crates/rpc/src/lib.rs [hunk 6]
```diff
@@ -129,7 +129,11 @@ impl Handler {
 
     async fn accounts(_: Params, _: Ctx) -> jsonrpc_core::Result<serde_json::Value> {
         let wallets = Wallets::read().await;
-        let address = wallets.get_current_wallet().get_current_address().await;
+        let address = wallets
+            .get_current_wallet()
+            .ok_or(Error::Wallets(iron_wallets::Error::NoWallet))?
```
Comment (naps62) [inline crates/rpc/src/lib.rs:132]: I believe this should work, no?

```suggestion
            .ok_or(iron_wallets::Error::NoWallet)?
```

File: gui/src/components/Onboarding/Carousel.tsx (hunk 4 above)
Comment (naps62) [reply gui/src/components/Onboarding/Carousel.tsx]: well, what about just removing those buttons altogether?
I think we discussed this at some point (though as a separate task), that you shouldn't be able to hop forward/backwards in the flow, as that may mess things up

Comment (pferreira101) [reply gui/src/components/Onboarding/Carousel.tsx]: By removing the buttons how are we supposed to advance to the next step/progress on the onboard? Don't know how to proceed with this 🤔

File: crates/rpc/src/lib.rs (hunk 6 above)
Comment (pferreira101) [reply crates/rpc/src/lib.rs:132]: Added this on the rpc crate:
```
#[error(transparent)]
Wallets(#[from] iron_wallets::Error)
```
`.ok_or(iron_wallets::Error::NoWallet)?` did not work
Tried other ways to make it less verbose but none of them compiled
